src/lib/sql/sql_translator.cpp

```diff
@@ -854,7 +853,27 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_predicate(
   };
 
   // TODO(anybody): handle IN with join
-  auto scan_type = translate_operator_type_to_scan_type(hsql_expr.opType);
+
+  auto predicate_negated = (hsql_expr.opType == hsql::kOpNot);
+
+  const auto *column_ref_hsql_expr = hsql_expr.expr;
+  ScanType scan_type;
+
+  if (predicate_negated) {
+    Assert(hsql_expr.expr != nullptr, "NOT operator without further expressions");
+    scan_type = translate_operator_type_to_scan_type(hsql_expr.expr->opType);
+
+    Assert(scan_type == ScanType::OpIsNull, "Only IS NULL can be negated");
```

Comment:
"currently"?

Reply:
This is a weird thing with the SQL parser. So far, IS NULL is the only operator type I could find that has a separate layer of `Expr` when it's negated. But I'm not sure if there might be others

Reply:
You can generally do `NOT expr`, e.g., `WHERE NOT a < 3`. I can see how the sql-parser treats those differently, because one is in front of the expr, while the other is part of the keyword `IS NOT NULL`.

The other one could somewhat easily be implemented by negating the operators inside. I'll put that on the postponed list.

That discussion should be in the comment so that we don't confuse anyone who wants to use/implement `NOT expr`.

Reply:
Ah I see, I didnt think about `NOT` being used like that. I added a comment to clarify this.